pyam/core.py

```diff
@@ -940,18 +946,25 @@ def categorize(
             assign a linestyle to this category for plotting
         """
         # add plotting run control
+
         for kind, arg in [
             ("color", color),
             ("marker", marker),
             ("linestyle", linestyle),
         ]:
             if arg:
                 run_control().update({kind: {name: {value: arg}}})
-        # find all data that matches categorization
-        rows = _apply_criteria(self._data, criteria, in_range=True, return_test="all")
-        idx = make_index(rows, cols=self.index.names)
 
-        if len(idx) == 0:
+        # find all data that matches categorization
+        # TODO: if validate returned an empty index, this check would be easier
+        not_valid = self.validate(criteria=criteria, **kwargs)
```

Comment:
Here @danielhuppmann the validate kwargs like `upper_bound` and other filtering args are taken in